### PR TITLE
port tendermint lc to union-ibc

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**

### CODE REVIEW

**Review by PoisonPhang — APPROVED** [review]

File: lib/tendermint-light-client-types/Cargo.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ name    = "tendermint-light-client-types"
 version = "0.1.0"
 
 [dependencies]
+alloy                    = { workspace = true, features = ["sol-types"] }
```
Comment (benluelo) [inline lib/tendermint-light-client-types/Cargo.toml:7]: feature gate this to `ethabi` and make optional

File: lib/tendermint-light-client-types/Cargo.toml [hunk 2]
```diff
@@ -12,7 +13,9 @@ unionlabs                = { workspace = true }
 
 [features]
 default = ["proto"]
+ethabi  = ["unionlabs/ethabi"]
 proto   = ["unionlabs/proto", "cometbft-types/proto"]
+serde   = []
```
Comment (benluelo) [inline lib/tendermint-light-client-types/Cargo.toml:18]: this should include `dep:serde`

File: lib/tendermint-light-client-types/Cargo.toml [hunk 3]
```diff
@@ -4,6 +4,7 @@ name    = "tendermint-light-client-types"
 version = "0.1.0"
 
 [dependencies]
+alloy                    = { workspace = true, features = ["sol-types"] }
 cometbft-types.workspace = true
 protos                   = { workspace = true, features = ["ibc+lightclients+tendermint+v1"] }
 serde                    = { workspace = true, features = ["derive"] }
```
Comment (benluelo) [inline lib/tendermint-light-client-types/Cargo.toml:10]: make optional

File: lib/tendermint-light-client-types/Cargo.toml [hunk 4]
```diff
@@ -4,6 +4,7 @@ name    = "tendermint-light-client-types"
 version = "0.1.0"
 
 [dependencies]
+alloy                    = { workspace = true, features = ["sol-types"] }
 cometbft-types.workspace = true
 protos                   = { workspace = true, features = ["ibc+lightclients+tendermint+v1"] }
```
Comment (benluelo) [inline lib/tendermint-light-client-types/Cargo.toml:9]: make optional, feature gate under `proto`

**Review by benluelo — DISMISSED** [review]

**Pull request merged**